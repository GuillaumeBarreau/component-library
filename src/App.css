.App {
  text-align: center;
}

.App-section--container {
  display: flex;
  min-height: 100vh;
  width: cal(100% - 4rem);
  align-items: center;
  justify-content: center;
  flex-direction: column;
  width: 100%;
}

.App-section--container__content {
  display: flex;
  flex-flow: wrap;
  width: 100%;
  justify-content: center;
  margin: 1rem;
}

.App-header {
  opacity: 1;
  background-size: 8px 8px;
  background-image: repeating-linear-gradient(
    45deg,
    #00002e 0,
    #202038 0.8px,
    #000000 0,
    #000000 50%
  );

  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
}

.App-link {
  color: #61dafb;
}
