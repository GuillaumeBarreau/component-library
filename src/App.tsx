import "./App.css";
import React, { useState, useEffect } from "react";
import { Atoms, Molecules, Organisms } from "./components/";

function App() {
  const [InputTextValue, setInputTextValue] = useState<string>("");
  const [userLogin, setUserLogin] = useState<boolean>(false);
  const [running, setRunning] = useState(false);
  const [progress, setProgress] = useState(0);
  let interval: NodeJS.Timeout | any = undefined;

  const inputEventOnChange = (value: React.SetStateAction<string>) => {
    setInputTextValue(value);
  };

  const buttonEventClick = () => {
    setUserLogin(true);
  };

  const noticesBase = ["default", "success", "error", "warning", "info"];
  const noticesBase2 = [...noticesBase, "disabled"];
  // const noticesBase3 = [
  //   ...noticesBase,
  //   ...noticesBase2,
  //   "primary",
  //   "secondary",
  // ];

  const sizesBase = ["xsmall", "small", "medium"];
  const sizesBase2 = ["xsmall", "small", "default", "medium"];

  useEffect(() => {
    if (running) {
      interval = setInterval(() => {
        setProgress((prev) => prev + 1);
      }, 10);
    } else {
      clearInterval(interval);
    }
  }, [running]);

  useEffect(() => {
    if (progress === 100) {
      setRunning(false);
      clearInterval(interval);
    }
  }, [progress]);

  const NodeLogin = (
    <>
      <Atoms.Input
        handleOnChange={(e) => inputEventOnChange(e.target.value)}
        placeholder="login"
        namedInput="login"
        labelDescription="login"
        value={InputTextValue}
      />
      <Atoms.Button
        handleClick={buttonEventClick}
        notice={InputTextValue ? "info" : "disabled"}
      >
        disabled
      </Atoms.Button>
    </>
  );

  const NodeDashboard = null;

  return (
    <div className="App">
      <header className="App-header">
        <section className="App-section--container">
          <div className="App-section--container__content">
            <>
              {sizesBase2.map((sizeValue: any) => {
                return (
                  <div
                    key={sizeValue}
                    className="App-section--container__content"
                  >
                    {noticesBase.map(
                      (value: any): JSX.Element => (
                        <Atoms.Button
                          key={value}
                          handleClick={() => console.log("here")}
                          notice={value}
                          size={sizeValue}
                        >
                          {value}
                        </Atoms.Button>
                      )
                    )}
                  </div>
                );
              })}
            </>
          </div>

          <br />

          <div className="App-section--container__content">
            <>
              {noticesBase2.map(
                (value: any): JSX.Element => (
                  <Atoms.Button
                    key={value}
                    handleClick={() => console.log("here")}
                    isBlock
                    notice={value}
                  >
                    {value}
                  </Atoms.Button>
                )
              )}
            </>
          </div>

          <br />

          <div className="App-section--container__content">
            <>
              {noticesBase2.map(
                (value: any): JSX.Element => (
                  <Atoms.Button
                    key={value}
                    handleClick={() => console.log("here")}
                    notice={value}
                    variant="outlined"
                  >
                    {value}
                  </Atoms.Button>
                )
              )}
            </>
          </div>

          <br />

          <div className="App-section--container__content">
            <>
              {noticesBase2.map(
                (value: any): JSX.Element => (
                  <Atoms.Button
                    key={value}
                    handleClick={() => console.log("here")}
                    notice={value}
                    variant="text"
                  >
                    {value}
                  </Atoms.Button>
                )
              )}
            </>
          </div>
        </section>

        <br />

        <section className="App-section--container">
          <div className="App-section--container__content">
            <>
              {sizesBase2.map((sizeValue: any) => {
                return (
                  <div
                    key={sizeValue}
                    className="App-section--container__content"
                  >
                    {noticesBase.map(
                      (value: any): JSX.Element => (
                        <Atoms.Button
                          key={value}
                          handleClick={() => console.log("here")}
                          notice={value}
                          size={sizeValue}
                          iconName="coffee"
                        >
                          {value}
                        </Atoms.Button>
                      )
                    )}
                  </div>
                );
              })}
            </>
          </div>

          <br />

          <div className="App-section--container__content">
            <>
              {noticesBase2.map(
                (value: any): JSX.Element => (
                  <Atoms.Button
                    key={value}
                    handleClick={() => console.log("here")}
                    isBlock
                    iconName="coffee"
                    notice={value}
                  >
                    {value}
                  </Atoms.Button>
                )
              )}
            </>
          </div>

          <br />

          <div className="App-section--container__content">
            <>
              {noticesBase2.map(
                (value: any): JSX.Element => (
                  <Atoms.Button
                    key={value}
                    handleClick={() => console.log("here")}
                    notice={value}
                    variant="outlined"
                    iconName="coffee"
                  >
                    {value}
                  </Atoms.Button>
                )
              )}
            </>
          </div>

          <br />

          <div className="App-section--container__content">
            <>
              {noticesBase2.map(
                (value: any): JSX.Element => (
                  <Atoms.Button
                    key={value}
                    handleClick={() => console.log("here")}
                    notice={value}
                    variant="text"
                    iconName="coffee"
                  >
                    {value}
                  </Atoms.Button>
                )
              )}
            </>
          </div>
        </section>

        <section className="App-section--container">
          <>
            {sizesBase.map((sizeValue: any) => {
              return (
                <div
                  key={sizeValue}
                  className="App-section--container__content"
                >
                  {noticesBase.map(
                    (value: any): JSX.Element => (
                      <Atoms.Badge key={value} size={sizeValue} notice={value}>
                        {value}
                      </Atoms.Badge>
                    )
                  )}
                </div>
              );
            })}
          </>

          <br />

          <>
            {sizesBase.map((sizeValue: any) => {
              return (
                <div
                  key={sizeValue}
                  className="App-section--container__content"
                >
                  {noticesBase.map(
                    (value: any): JSX.Element => (
                      <Atoms.Badge
                        key={value}
                        iconName="circle-exclamation"
                        size={sizeValue}
                        notice={value}
                      >
                        {value}
                      </Atoms.Badge>
                    )
                  )}
                </div>
              );
            })}
          </>
        </section>

        <section className="App-section--container">
          {(progress === 100 && (
            <Atoms.Button
              handleClick={() => {
                setRunning(false);
                setProgress(0);
              }}
              notice="default"
            >
              Reset ProgressBar
            </Atoms.Button>
          )) || (
            <Atoms.Button
              handleClick={() => setRunning(!running)}
              notice="default"
            >
              {running ? "Stop ProgressBar" : "Start ProgressBar"}
            </Atoms.Button>
          )}
          <br />
          <>
            {noticesBase.map(
              (value: any): JSX.Element => (
                <div key={value} className="App-section--container__content">
                  <Atoms.ProgressBar progress={progress} notice={value} />
                </div>
              )
            )}
          </>
          <br />
          <>
            {sizesBase.map(
              (value: any): JSX.Element => (
                <div key={value} className="App-section--container__content">
                  <Atoms.ProgressBar progress={progress} size={value} />
                </div>
              )
            )}
          </>
        </section>

        <section className="App-section--container">
          <>
            {noticesBase.map(
              (value: any): JSX.Element => (
                <div key={value} className="App-section--container__content">
                  <Organisms.Alert notice={value}>
                    Lorem ipsum dolor, sit amet consectetur adipisicing elit.
                    Vel ratione nostrum, molestiae fugit vero consectetur?
                  </Organisms.Alert>
                </div>
              )
            )}
          </>
        </section>

        <section className="App-section--container">
          <>
            {noticesBase.map(
              (value: any): JSX.Element => (
                <div key={value} className="App-section--container__content">
                  <Organisms.Alert notice={value} dismissible>
                    Lorem ipsum dolor, sit amet consectetur adipisicing elit.
                    Vel ratione nostrum, molestiae fugit vero consectetur?
                  </Organisms.Alert>
                </div>
              )
            )}
          </>
        </section>

        <section className="App-section--container">
          {(progress === 100 && (
            <Atoms.Button
              handleClick={() => {
                setRunning(false);
                setProgress(0);
              }}
              notice="default"
            >
              Reset ProgressBar
            </Atoms.Button>
          )) || (
            <Atoms.Button
              handleClick={() => setRunning(!running)}
              notice="default"
            >
              {running ? "Stop ProgressBar" : "Start ProgressBar"}
            </Atoms.Button>
          )}
          <br />
          <>
            {noticesBase.map(
              (value: any): JSX.Element => (
                <div key={value} className="App-section--container__content">
                  <Organisms.Alert
                    notice={value}
                    progressBar={progress}
                    dismissible
                  >
                    Lorem ipsum dolor, sit amet consectetur adipisicing elit.
                    Vel ratione nostrum, molestiae fugit vero consectetur?
                  </Organisms.Alert>
                </div>
              )
            )}
          </>
        </section>

        <section className="App-section--container">
          {(progress === 100 && (
            <Atoms.Button
              handleClick={() => {
                setRunning(false);
                setProgress(0);
              }}
              notice="default"
            >
              Reset ProgressBar
            </Atoms.Button>
          )) || (
            <Atoms.Button
              handleClick={() => setRunning(!running)}
              notice="default"
            >
              {running ? "Stop ProgressBar" : "Start ProgressBar"}
            </Atoms.Button>
          )}
          <br />
          <>
            {noticesBase.map(
              (value: any): JSX.Element => (
                <div key={value} className="App-section--container__content">
                  <Organisms.Alert
                    notice={value}
                    progressBar={progress}
                    dismissible
                    progressBarLabel
                  >
                    Lorem ipsum dolor, sit amet consectetur adipisicing elit.
                    Vel ratione nostrum, molestiae fugit vero consectetur?
                  </Organisms.Alert>
                </div>
              )
            )}
          </>
        </section>

        <section className="App-section--container">
          {userLogin ? NodeDashboard : NodeLogin}
        </section>
      </header>
    </div>
  );
}

export default App;
