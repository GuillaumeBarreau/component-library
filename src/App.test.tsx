import React from 'react';
import { render, screen } from '@testing-library/react';
import { shallow } from 'enzyme';
import App from './App';

it('renders correctly', () => {
  const wrapper = shallow(
    <App/>
  );
  expect(wrapper.debug()).toMatchSnapshot();
});

it('mount correctly', () => {
  const wrapper = shallow(
    <App />
  );
  expect(() => wrapper).not.toThrow();
});
