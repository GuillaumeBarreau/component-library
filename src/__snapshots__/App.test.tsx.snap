// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
"<div className=\\"App\\">
  <header className=\\"App-header\\">
    <section className=\\"App-section--container\\">
      <div className=\\"App-section--container__content\\">
        <Button handleClick={[Function: handleClick]} size=\\"xsmall\\">
          default
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"success\\" size=\\"xsmall\\">
          success
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"info\\" size=\\"xsmall\\">
          info
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"warning\\" size=\\"xsmall\\">
          warning
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"error\\" size=\\"xsmall\\">
          error
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"disabled\\" size=\\"xsmall\\">
          disabled
        </Button>
      </div>
      <div className=\\"App-section--container__content\\">
        <Button handleClick={[Function: handleClick]} size=\\"small\\">
          default
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"success\\" size=\\"small\\">
          success
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"info\\" size=\\"small\\">
          info
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"warning\\" size=\\"small\\">
          warning
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"error\\" size=\\"small\\">
          error
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"disabled\\" size=\\"small\\">
          disabled
        </Button>
      </div>
      <div className=\\"App-section--container__content\\">
        <Button handleClick={[Function: handleClick]}>
          default
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"success\\">
          success
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"info\\">
          info
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"warning\\">
          warning
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"error\\">
          error
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"disabled\\">
          disabled
        </Button>
      </div>
      <div className=\\"App-section--container__content\\">
        <Button handleClick={[Function: handleClick]} size=\\"medium\\">
          default
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"success\\" size=\\"medium\\">
          success
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"info\\" size=\\"medium\\">
          info
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"warning\\" size=\\"medium\\">
          warning
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"error\\" size=\\"medium\\">
          error
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"disabled\\" size=\\"medium\\">
          disabled
        </Button>
      </div>
      <div className=\\"App-section--container__content\\">
        <Button handleClick={[Function: handleClick]} isBlock={true}>
          default Block
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"success\\" isBlock={true}>
          success Block
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"info\\" isBlock={true}>
          info Block
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"warning\\" isBlock={true}>
          warning Block
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"error\\" isBlock={true}>
          error Block
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"disabled\\" isBlock={true}>
          disabled Block
        </Button>
      </div>
    </section>
    <section className=\\"App-section--container\\">
      <div className=\\"App-section--container__content\\">
        <Button handleClick={[Function: handleClick]} size=\\"xsmall\\" iconName=\\"fa-coffee\\">
          default
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"success\\" size=\\"xsmall\\" iconName=\\"fa-coffee\\">
          success
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"info\\" size=\\"xsmall\\" iconName=\\"fa-coffee\\">
          info
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"warning\\" size=\\"xsmall\\" iconName=\\"fa-coffee\\">
          warning
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"error\\" size=\\"xsmall\\" iconName=\\"fa-coffee\\">
          error
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"disabled\\" size=\\"xsmall\\" iconName=\\"fa-coffee\\">
          disabled
        </Button>
      </div>
      <div className=\\"App-section--container__content\\">
        <Button handleClick={[Function: handleClick]} size=\\"small\\" iconName=\\"fa-coffee\\">
          default
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"success\\" size=\\"small\\" iconName=\\"fa-coffee\\">
          success
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"info\\" size=\\"small\\" iconName=\\"fa-coffee\\">
          info
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"warning\\" size=\\"small\\" iconName=\\"fa-coffee\\">
          warning
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"error\\" size=\\"small\\" iconName=\\"fa-coffee\\">
          error
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"disabled\\" size=\\"small\\" iconName=\\"fa-coffee\\">
          disabled
        </Button>
      </div>
      <div className=\\"App-section--container__content\\">
        <Button handleClick={[Function: handleClick]} iconName=\\"fa-coffee\\">
          default
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"success\\" iconName=\\"fa-coffee\\">
          success
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"info\\" iconName=\\"fa-coffee\\">
          info
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"warning\\" iconName=\\"fa-coffee\\">
          warning
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"error\\" iconName=\\"fa-coffee\\">
          error
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"disabled\\" iconName=\\"fa-coffee\\">
          disabled
        </Button>
      </div>
      <div className=\\"App-section--container__content\\">
        <Button handleClick={[Function: handleClick]} size=\\"medium\\" iconName=\\"fa-coffee\\">
          default
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"success\\" size=\\"medium\\" iconName=\\"fa-coffee\\">
          success
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"info\\" size=\\"medium\\" iconName=\\"fa-coffee\\">
          info
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"warning\\" size=\\"medium\\" iconName=\\"fa-coffee\\">
          warning
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"error\\" size=\\"medium\\" iconName=\\"fa-coffee\\">
          error
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"disabled\\" size=\\"medium\\" iconName=\\"fa-coffee\\">
          disabled
        </Button>
      </div>
      <div className=\\"App-section--container__content\\">
        <Button handleClick={[Function: handleClick]} isBlock={true} iconName=\\"fa-coffee\\">
          default Block
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"success\\" isBlock={true} iconName=\\"fa-coffee\\">
          success Block
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"info\\" isBlock={true} iconName=\\"fa-coffee\\">
          info Block
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"warning\\" isBlock={true} iconName=\\"fa-coffee\\">
          warning Block
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"error\\" isBlock={true} iconName=\\"fa-coffee\\">
          error Block
        </Button>
        <Button handleClick={[Function: handleClick]} notice=\\"disabled\\" isBlock={true} iconName=\\"fa-coffee\\">
          disabled Block
        </Button>
      </div>
    </section>
    <section className=\\"App-section--container\\">
      <div className=\\"App-section--container__content\\">
        <Badge size=\\"xsmall\\">
          default
        </Badge>
        <Badge notice=\\"success\\" size=\\"xsmall\\">
          success
        </Badge>
        <Badge notice=\\"info\\" size=\\"xsmall\\">
          info
        </Badge>
        <Badge notice=\\"warning\\" size=\\"xsmall\\">
          warning
        </Badge>
        <Badge notice=\\"error\\" size=\\"xsmall\\">
          error
        </Badge>
      </div>
      <div className=\\"App-section--container__content\\">
        <Badge size=\\"small\\">
          default
        </Badge>
        <Badge notice=\\"success\\" size=\\"small\\">
          success
        </Badge>
        <Badge notice=\\"info\\" size=\\"small\\">
          info
        </Badge>
        <Badge notice=\\"warning\\" size=\\"small\\">
          warning
        </Badge>
        <Badge notice=\\"error\\" size=\\"small\\">
          error
        </Badge>
      </div>
      <div className=\\"App-section--container__content\\">
        <Badge>
          default
        </Badge>
        <Badge notice=\\"success\\">
          success
        </Badge>
        <Badge notice=\\"info\\">
          info
        </Badge>
        <Badge notice=\\"warning\\">
          warning
        </Badge>
        <Badge notice=\\"error\\">
          error
        </Badge>
      </div>
    </section>
    <section className=\\"App-section--container\\">
      <div className=\\"App-section--container__content\\">
        <Question>
          Molecules.Question
        </Question>
        <Answer handleClick={[Function: handleClick]}>
          Molecules.Answer
        </Answer>
      </div>
    </section>
    <section className=\\"App-section--container\\">
      <Input handleOnChange={[Function: handleOnChange]} placeholder=\\"login\\" namedInput=\\"login\\" labelDescription=\\"login\\" value=\\"\\" />
      <Button handleClick={[Function: buttonEventClick]} notice=\\"disabled\\">
        disabled
      </Button>
    </section>
  </header>
</div>"
`;
